sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/ConfigurationClientBuilder.java

```diff
@@ -145,38 +149,55 @@ public ConfigurationClient buildClient() {
      * @throws IllegalStateException If {@link #connectionString(String) connectionString} has not been set.
      */
     public ConfigurationAsyncClient buildAsyncClient() {
+        // Global Env configuration store
         Configuration buildConfiguration =
             (configuration == null) ? Configuration.getGlobalConfiguration().clone() : configuration;
-        ConfigurationClientCredentials configurationCredentials = getConfigurationCredentials(buildConfiguration);
-        String buildEndpoint = getBuildEndpoint(configurationCredentials);
 
+        // Service version
+        ConfigurationServiceVersion serviceVersion =
+            version != null ? version : ConfigurationServiceVersion.getLatest();
+
+        // AAD token credential
+        TokenCredential tokenCredentialEnvVar = tokenCredential;
+        // Connection string credential
+        ConfigurationClientCredentials configurationCredentials = null;
+
+        // Endpoint
+        String buildEndpoint = endpoint;
+        if (tokenCredentialEnvVar == null) {
+            configurationCredentials = getConfigurationCredentials(buildConfiguration);
```

Comment:
Honestly, I don't think we need to try and get configuration credentials from an environment variable and it should be removed entirely. (the AZURE_APPCONFIG_CONNECTION_STRING)

iirc, this existed before I owned the code. I think it's easy enough for consumers to perform their own Sys.getEnv("") and pass it into `connectionString(String)`. Does any other language offer this convenience environment variable?

Reply:
Agreed. We should not looking for Connection String for the user